patternkit.module

```diff
@@ -69,28 +69,38 @@ function patternkit_flush_metadata_cache() {
  *   An array of Pattern Kit Library objects.
  */
 function patternkit_pattern_libraries() {
-  // @todo Add caching.
-  return module_invoke_all('patternkit_library');
+  static $libraries;
+
+  // If undefined, create the libraries metadata.
+  if (is_null($libraries)) {
+    if ($cache = cache_get('patternkit_libraries')) {
+      $libraries = $cache->data;
+    }
+    else {
+      $libraries = module_invoke_all('patternkit_library');
+
+      if (!empty($libraries)) {
+        // Cache for one day.
+        cache_set('patternkit_library', $libraries, time() + 60 * 60 * 24);
+      }
+    }
+  }
+
+  // Triggers initialization of the Twig wrapper.
+  PatternkitTwigWrapper::getInstance($libraries);
+
+  return $libraries;
 }
 
 /**
  * Implements hook_patternkit_library().
  */
 function patternkit_patternkit_library() {
-  // @todo Replace with a Service that calls a Factory.
```

Comment:
Should we be removing support for themes to add pattern libraries? Maybe make it an option?

My reasoning is in Drupal 8, adding the 'namespace' key to the theme is the generally accepted way to add component libraries via the Component Libraries module.

Reply:
If it sounds good to you, I'll get a patch going for the following:

- Populate libraries from both hook usage as well as theme namespaces
- Set the default 'enabled' state of each library based on module/theme 'enabled' state
- Allow overriding the 'enabled' state of each library in the admin UI / via variables

Reply:
I replaced the namespace discovery stuff. no reason not to allow both.